### PR TITLE
Refactors and move InMemoryCrud Repo implementation to common

### PR DESCRIPTION
**Pull request by mdanish98**
## Description of Changes

Moves all the InMemory CrudRepository implementation to common.
It is the important refactoring for the ticket #437

### CODE REVIEW

File: basyx.aasrepository/basyx.aasrepository-backend/pom.xml [hunk 1]
```diff
@@ -19,6 +19,10 @@
 			<groupId>org.eclipse.digitaltwin.basyx</groupId>
 			<artifactId>basyx.aasrepository-core</artifactId>
 		</dependency>
+		<dependency>
+			<groupId>org.eclipse.digitaltwin.basyx</groupId>
+			<artifactId>basyx.mongodbcore</artifactId>
```
Comment (mateusmolina-iese) [inline basyx.aasrepository/basyx.aasrepository-backend/pom.xml:24]: Why basyx.mongodbcore in aasrepository-backend?

**Review by mateusmolina-iese — CHANGES_REQUESTED** [review]
Any reason to leave out the AASXFileServer?

File: basyx.aasrepository/basyx.aasrepository-backend/pom.xml (hunk 1 above)
Comment (mdanish98) [reply basyx.aasrepository/basyx.aasrepository-backend/pom.xml]: It was added by mistake, removed it.

**Review by mateusmolina-iese — APPROVED** [review]
lgtm!

**Pull request merged**